### PR TITLE
boards: stm32: Move STM32 boards SPI pin configuration to device tree

### PR DESCRIPTION
**Pull request by erwango**
Move SPI pin configuration to device tree for all STM32 based boards.
This should not bring any change in pin configuration.
    
Though, some adjuments have been made on following boards for coherency:
 * b_l4s5i_iot01a1: Removed "very-high-speed" on SPI3 MOSI pin.
    Tested with no impact.
 * black_f407zg_pro: SPI1 pins configured but not SPI1 node.
    Removed pin config as not documented neither.
 * sensortile_box: SPI2 pins configured but not SPI2 node.
    Removed pin config as not documented neither.
 * nucleo_g431rb/nucleo_g474re/nucleo_l476rg/nucleo_l4r5zi:
    SPI2 and SPI3 pins were configured but missing.
    Added since they were documented.

Based on https://github.com/zephyrproject-rtos/zephyr/pull/29231

### CODE REVIEW

File: boards/arm/nucleo_l053r8/nucleo_l053r8.dts [hunk 1]
```diff
@@ -66,6 +66,7 @@
 };
 
 &spi1 {
+	pinctrl-0 = <&spi1_sck_pa5 &spi1_miso_pa6 &spi1_mosi_pa7>;
```
Comment (ABOSTM) [inline boards/arm/nucleo_l053r8/nucleo_l053r8.dts:69]: Pin added in DT, but I can't see removal from pinmux

File: boards/arm/stm32l1_disco/pinmux.c [hunk 2]
```diff
@@ -14,17 +14,17 @@
...
 #endif
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(spi2), okay) && CONFIG_SPI
-#ifdef CONFIG_SPI_STM32_USE_HW_SS
+
 	{STM32_PIN_PB12, STM32L1X_PINMUX_FUNC_PB12_SPI2_NSS},
-#endif /* CONFIG_SPI_STM32_USE_HW_SS */
+
 	{STM32_PIN_PB13, STM32L1X_PINMUX_FUNC_PB13_SPI2_SCK},
 	{STM32_PIN_PB14, STM32L1X_PINMUX_FUNC_PB14_SPI2_MISO},
 	{STM32_PIN_PB15, STM32L1X_PINMUX_FUNC_PB15_SPI2_MOSI},
```
Comment (ABOSTM) [inline boards/arm/stm32l1_disco/pinmux.c:30]: To be removed, as defined in DT

**Review by ABOSTM — CHANGES_REQUESTED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by sidcha — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

File: boards/arm/nucleo_f756zg/pinmux.c [hunk 3]
```diff
@@ -36,9 +36,9 @@ static const struct pin_config pinconf[] = {
 	{ STM32_PIN_PA12, STM32F7_PINMUX_FUNC_PA12_OTG_FS_DP },
 #endif	/* CONFIG_USB_DC_STM32 */
 #if DT_NODE_HAS_STATUS(DT_NODELABEL(spi1), okay) && CONFIG_SPI
```
Comment (galak) [inline boards/arm/nucleo_f756zg/pinmux.c:38]: shouldn't the pins be removed here?

Comment (erwango) [reply boards/arm/nucleo_f756zg/pinmux.c:38]: Right txs, let me fix this.

Comment (erwango) [reply boards/arm/nucleo_f756zg/pinmux.c]: Done

**Review by galak — APPROVED** [review]

**Review by lochej — APPROVED** [review]

**Comment by galak** [comment]
Removed BK commit and merged.

**Pull request merged**